io/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -149,14 +145,14 @@ private object RTS {
     if (!stack.isEmpty) stack.pop()(value).asInstanceOf[IO[E, Any]] else null
 
   object Catcher extends Function[Any, IO[Any, Any]] {
-    final def apply(v: Any): IO[Any, Any] = IO.now(\/-(v))
+    final def apply(v: Any): IO[Any, Any] = IO.now(Right(v))
```

Comment:
This is not the correct change. The user will construct the value based on success, so we'll need an elaborate surgical operation here. 😆 

```scala
class Catcher(err: Any => IO[Any, Any], suc: Any => IO[Any, Any]) extends (Any => IO[Any, Any]) {
  final def apply(v: Any): IO[Any, Any] = suc(v)
}
```

Unfortunately this will necessitate many other changes in the RTS. I think we should pair program on this task unless you're feeling crazy ambitious today. 😉

Reply:
I do feel crazy ambitious most of the time, but not today, it's 1.30am here in Sydney :) Although the offer to pair program is really tempting i must say

Reply:
I was almost certain it wasn't the right replacement, I did it just to get it compiling. All the `Any` `Any` `Any` `Any` were not of great help here :smile: